doc/ebpf/env.rst

```diff
@@ -0,0 +1,630 @@
+Building Linux Image for Working with eBPF in QEMU
+==================================================
+
+.. contents:: :local:
+
+Overview
+--------
+
+This is a comprehensive guide about creating an environment for 
+building, running and debugging eBPF programs for ARC processors using
+`GNU toolchain <https://github.com/foss-for-synopsys-dwc-arc-processors/arc-gnu-toolchain>`_
+and `QEMU <https://github.com/foss-for-synopsys-dwc-arc-processors/qemu>`_. Though we consider
+ARC HS 3x/4x on QEMU as a reference platform, the same guide is applicable for boards
+like HS Development Kit.
+
+This guide consists of these steps:
+
+1. Preparing your Linux host for building Buildroot image and Linux kernel and for building third-party tools and libraries.
+
+2. Building and installing third-party tools and libraries: ``elfutils``, ``pahole`` and ``bpftool``. We are going to build them manually to ensure that the latest versions are used.
+
+3. Preparing building environment: cloning all necessary repositories, configuring SSH keys on host's side, etc.
+
+4. Building Buildroot image and Linux kernel.
+
+5. Building and running eBPF programs.
+
```

Comment:
If you can, add a link to each number (1,2,3,4,5) above. So one could jump right to that section in this doc.

This is more like a suggestion, I don't feel strong about it.

Reply:
There is contents above that text, so reader can use it for jumping to sections.